target/linux/ipq806x/base-files/etc/diag.sh

```diff
@@ -1,34 +1,41 @@
 #!/bin/sh
+# Copyright (C) 2016 Henryk Heisig hyniu@o2.pl
 
 . /lib/functions/leds.sh
 . /lib/ipq806x.sh
 
-get_status_led() {
-	case $(ipq806x_board_name) in
-	c2600)
-		status_led="c2600:blue:status"
-		;;
-	ea8500)
-		status_led="ea8500:white:power"
-		;;
-	esac
-}
+boot="$(ipq806x_get_dt_led_chosen boot)"
+failsafe="$(ipq806x_get_dt_led_chosen failsafe)"
+running="$(ipq806x_get_dt_led_chosen running)"
+upgrade="$(ipq806x_get_dt_led_chosen upgrade)"
 
 set_state() {
-	get_status_led
+	status_led="$boot"
 
 	case "$1" in
 	preinit)
 		status_led_blink_preinit
 		;;
 	failsafe)
+		status_led_off
+		status_led="$failsafe"
 		status_led_blink_failsafe
 		;;
 	preinit_regular)
 		status_led_blink_preinit_regular
 		;;
+	upgrade)
+		[ -n "$running" ] & {
+			status_led="$upgrade"
```

Comment:
It have to. Same typo that I've introduced with my lantiq patch back in the time...